backend/app/src/main/java/ch/zhaw/studyflow/domain/calendar/MemoryCalendarDao.java

```diff
@@ -0,0 +1,53 @@
+package ch.zhaw.studyflow.domain.calendar;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * In-memory implementation of the CalendarDao interface.
+ * This class provides methods to perform CRUD operations on Calendar objects
+ * stored in memory.
+ */
+public class MemoryCalendarDao implements CalendarDao {
+    private final Map<Long, Map<Long, Calendar>> userCalendars = new HashMap<>();
+
+    @Override
+    public Calendar create(Calendar calendar) {
+        userCalendars.computeIfAbsent(calendar.getOwnerId(), k -> new HashMap<>()).put(calendar.getId(), calendar);
```

Comment:
Maybe we should check for null here and set an id (auto-increment / simulate the database).